### PR TITLE
Feature/search table

### PR DESCRIPTION
**Pull request by peterpep**
added autocomplete search

### CODE REVIEW

File: src/components/Overview.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<template>
```
Comment (bry-guy) [inline src/components/Overview.vue]: move 'pages' to src/pages
- src/pages/chemicals/details (this is currently called 'overview')

leave shared components (such as dynamic table) in src/components

File: src/components/Overview.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <div>
+    <img style="width: 10%;" src="../assets/Logo_1.png">
+    <SearchTable :rows="this.$store.getters.chemicals" :columnNames="columns" @filterData="setData"></SearchTable>
```
Comment (bry-guy) [inline src/components/Overview.vue]: rename searcher

File: src/components/Overview.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  computed: {
+    columns: function() {
+      if (this.$store.getters.chemicals.length == 0)
+        return [];                    
+      return Object.keys(this.$store.getters.chemicals[0])
+    }
+  },
+  mounted: function() {
+    //api call here to get data
+    this.dataForTable = this.$store.getters.chemicals;
```
Comment (bry-guy) [inline src/components/Overview.vue]: rename dataForTable to dynamicTableDataSource

File: src/components/SearchTable.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<template>
+    <form @submit.prevent="$emit('filterData',searchJsonData(rows, searchIn, searchTerm))">
```
Comment (bry-guy) [inline src/components/SearchTable.vue]: change filterData to submitClicked

File: src/components/Overview.vue (hunk 2 above)
Comment (bry-guy) [inline src/components/Overview.vue]: rename filterData to 'submitClicked'
rename 'setData' to 'handleSubmitClicked'

File: src/components/Overview.vue [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    return {
+      dataForTable: [], //need to decide if this should be part of the store or just local placeholder variable      
+    }
+  },
+  components: {
+    DynamicTable,
+    SearchTable
+  },
+  methods: {
+    setData: function(e) {
```
Comment (bry-guy) [inline src/components/Overview.vue]: rename 'e' to 'filteredData'

File: src/components/SearchTable.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        <select class="dropDownEdges" v-model="searchIn">
```
Comment (bry-guy) [inline src/components/SearchTable.vue]: rename 'searchIn' to 'searchBy'

File: src/components/SearchTable.vue [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            getAutoCompleteData: function(inputData, column) {
+                var result = [];
+                if (column == "")
+                    column = 'id';
+                for (var i=0; i < inputData.length; i++) {
+                    result.push( {key: inputData[i][column].toString()});
+                }
+                return result;
+            },
+            formatColumn: function(name) {
```
Comment (bry-guy) [inline src/components/SearchTable.vue]: Make global helper function named 'splitOnUnderscores'

File: src/vuex/models/chemicals.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import {
+    USER_OBJECT
```
Comment (bry-guy) [inline src/vuex/models/chemicals.js]: rename USER_OBJECT to CHEMICAL_COLLECTION

**Review by bry-guy — CHANGES_REQUESTED** [review]
such code
many wow
very vue

**Comment by peterpep** [comment]
completed PR requested changes as well as removing the expiration status row highlighting from DynamicTable and moving into scoped slot (methods for binding class moved into mixin)

**Review by bry-guy — APPROVED** [review]

**Pull request merged**